validator/src/tendermint.rs

```diff
@@ -228,11 +228,62 @@ where
             return Err(ProtocolError::Abort);
         }
 
+        // Check if we want to upgrade the version.
+        // This assumes we only ever upgrade to the latest version
+        // and don't queue multiple upgrades.
```

Comment:
This is not a problem, I think. We can fix this if we ever want to do multiple upgrades at once.

Reply:
I agree with @hrxi . I left the comment in the code so that the (current) assumption is explicit. We can still change that in the future.